src/metabase/lib/equality.cljc

```diff
@@ -412,3 +412,21 @@
          (and (not (contains? matching nil))
               (= (count matching) (count columns))
               (every? #(= (count %) 1) (vals matching))))))
+
+(defn clauses-equal-ignoring-lib-uuids?
```

Comment:
New util function: decide whether two MBQL clauses are equal regardless of their `:lib/uuid`s. This is important for filter deduplication (see below)